Objects/longobject.c

```diff
@@ -4358,8 +4386,8 @@ static PyObject *
 long_neg(PyLongObject *v)
 {
     PyLongObject *z;
-    if (Py_ABS(Py_SIZE(v)) <= 1)
-        return PyLong_FromLong(-MEDIUM_VALUE(v));
+    if (IS_MEDIUM_VALUE(v))
+        return PyLong_FromLong(-medium_value(v));
```

Comment:
Similarly, I think this one can be `_PyLong_FromSDigit`, though I doubt it'll make any measurable difference.